src/hotspot/share/cds/cdsAccess.hpp

```diff
@@ -56,11 +56,12 @@ class CDSAccess : AllStatic {
 
   static void test_heap_access_api() NOT_CDS_JAVA_HEAP_RETURN;
 
-  static void* allocate_from_code_cache(size_t size) NOT_CDS_RETURN_(nullptr);
+  static void* allocate_from_code_cache(size_t size) NOT_CDS_JAVA_HEAP_RETURN_(nullptr);
```

Comment:
I think the following functions should be marked as `NOT_CDS_RETURN`. In the cpp file, they should be moved outside of the `INCLUDE_CDS_JAVA_HEAP` block:

```
void* CDSAccess::allocate_from_code_cache(size_t size)
size_t CDSAccess::get_cached_code_size()
void CDSAccess::set_cached_code_size(size_t sz)
bool CDSAccess::is_cached_code_region_empty()
bool CDSAccess::map_cached_code(ReservedSpace rs)
```

Reply:
yup, I agree